project/backend/src/main/java/com/google/sps/data/RestaurantScorer.java

```diff
@@ -14,21 +18,33 @@ private RestaurantScorer() {}
 
   /** Returns the percent match for a restaurant to a set of user preferences. */
   public static double score(Restaurant restaurant, JSONObject preferences) throws JSONException {
-    int preferredPriceLevel = preferences.getJSONObject("priceLevel").getInt("pref");
-    String preferredDiningExp = preferences.getJSONObject("diningExp").getString("pref");
     int priceLevelWeight = preferences.getJSONObject("priceLevel").getInt("weight");
     int diningExpWeight = preferences.getJSONObject("diningExp").getInt("weight");
-    int maxPoints = priceLevelWeight + diningExpWeight + 1;
+    int radiusWeight = preferences.getJSONObject("radius").getInt("weight");
+    int maxPoints = priceLevelWeight + diningExpWeight + radiusWeight + 1;
+    Map<String, Double> currLocation =
+        new Gson().fromJson(preferences.getJSONObject("currLocation").toString(), HashMap.class);
```

Comment:
Lets just use one static final instance of Gson: https://javadoc.io/doc/com.google.code.gson/gson/latest/com.google.gson/com/google/gson/Gson.html it says its thread-safe so we can safely reuse an instance